rest-api/rest-api.php

```diff
@@ -0,0 +1,672 @@
+
+            $post_id = $p_post_id;
+        }
+        else {
+            $post_id = $url_params['id'];
+        }
+
+        $url_params = $request->get_url_params();
+        $get_params = $request->get_query_params();
+        $silent = isset( $get_params['silent'] ) && $get_params['silent'] === 'true';
+
+        return DT_Posts::update_post( $url_params['post_type'], $post_id, $fields2, $silent );
+    }
+
+
+    public function check_field_value_exists( WP_REST_Request $request, string $maarifa_contact_id ) {
+        //Verify if maarifa_data already exists
+
+        dt_write_log( 'Check_field_value_exists' );
+
+        $params = $request->get_params();
+
+        if ( $maarifa_contact_id != null ) {
+
+            global $wpdb;
+            $result = $wpdb->get_results( $wpdb->prepare(
+                "SELECT `post_id`
+                            FROM $wpdb->postmeta
+                            WHERE meta_key LIKE `maarifa_data`
+                            AND meta_value = %s;", $maarifa_contact_id ) );
+
+
+            return $result;
+
+        }
+        return [];
+    }
+
+
+    public function add_interactions( WP_REST_Request $request ){
+
+        dt_write_log( 'Add_interactions' );
+
+        $url_params = $request->get_url_params();
+        $get_params = $request->get_query_params();
+        $body = $request->get_json_params() ?? $request->get_body_params();
+        $silent = isset( $get_params['silent'] ) && $get_params['silent'] === 'true';
+        $args = [];
+
+        $type = 'maarifa';
+
+        if ( !empty( $body ) ) {
+
+            $value = null;
+            $result = null;
+            $ret = null;
+
+            foreach ( $body as $key => $value ) {
+
+
+                if ( isset( $value['when_made'] ) ){
+                    $args['comment_date'] = $value['when_made'];
+                }
+
+                if ( isset( $value['responder_name'] ) ){
+                    $args['user_id'] = $value['responder_name'];
+                }
+
+                if ( isset( $value['notes'] ) ){
+
+                    $comment = 'Type: '. $type ."\n". 'Responder name: '. $value['responder_name']. "\n" .$value['notes'];
+
+                }
+
+                if ( isset( $value['meta'] ) ) {
+                    $args['comment_meta'] = $value['meta'];
+                }
+
+                if ( isset( $value['id'] ) )
+                {//If comment_id exists, update the comment
+
+                    $id = $value['id'];
+
+                    dt_write_log( 'Add_interactions UPDATE' );
+                    dt_write_log( $id );
+
+                    $result = DT_Posts::update_post_comment( $id, $comment, true, $type, $args );
+                }
+                else { //If doesn't, create a new comment
+                    dt_write_log( 'Add_interactions CREATE' );
+
+                    $result = DT_Posts::add_post_comment( $url_params['post_type'], $url_params['id'], $comment, $type, $args, true, $silent );
```

Comment:
I'm getting a date format error trying to save a comment/interaction. I'm guessing you haven't tested this yet, but just in case, here is the data I sent and response I received:

<img width="697" alt="image" src="https://github.com/user-attachments/assets/d8b82e6e-c1be-4169-b647-30bb225d07a2" />